### PR TITLE
Add integration test for windows platform

### PR DESCRIPTION
**Pull request by richardpen**
### Summary

Added integration test on windows:
* TestStartStopUnpulledImage
* TestStartStopUnpulledImageDigest
* TestPortForward
* TestMultiplePortForwards
* TestDynamicPortForward
* TestMultipleDynamicPortForward
* TestVolumesFrom
* TestVolumesFromRO

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/engine_integ_test.go [hunk 1]
```diff
@@ -125,6 +127,26 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
 		"Expected task to be STOPPED")
 }
 
+func dialWithRetries(proto string, address string, tries int, timeout time.Duration) (net.Conn, error) {
```
Comment (aaithal) [inline agent/engine/engine_integ_test.go]: why not use the `utils.SimpleBackoff()` instead?

File: agent/engine/engine_integ_test.go [hunk 2]
```diff
@@ -125,6 +127,26 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
...
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return conn, err
+}
+
+func removeImage(img string) {
+	removeEndpoint := utils.DefaultIfBlank(os.Getenv(DockerEndpointEnvVariable), DockerDefaultEndpoint)
+	client, _ := docker.NewClient(removeEndpoint)
```
Comment (aaithal) [inline agent/engine/engine_integ_test.go]: we're ok with silently consuming the error because?

File: agent/engine/engine_integ_test.go [hunk 3]
```diff
@@ -125,6 +127,26 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
...
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return conn, err
+}
+
+func removeImage(img string) {
+	removeEndpoint := utils.DefaultIfBlank(os.Getenv(DockerEndpointEnvVariable), DockerDefaultEndpoint)
```
Comment (aaithal) [inline agent/engine/engine_integ_test.go]: why not just use the `endpoint` var?

File: agent/engine/engine_windows_integ_test.go [hunk 4]
```diff
@@ -89,3 +104,392 @@ func createTestHealthCheckTask(arn string) *api.Task {
...
+	containerMap, _ := taskEngine.(*DockerTaskEngine).state.ContainerMapByArn(testTask.Arn)
+	cid1 := containerMap[testTask.Containers[0].Name].DockerID
+	cid2 := containerMap[testTask.Containers[1].Name].DockerID
+
+	cip1, err := getContainerIP(client, cid1)
+	require.NoError(t, err, "failed to acquire the container ip from docker")
+	cip2, err := getContainerIP(client, cid2)
+	require.NoError(t, err, "failed to acquire the container ip from docker")
+
+	time.Sleep(50 * time.Millisecond) // wait for Docker
```
Comment (aaithal) [inline agent/engine/engine_windows_integ_test.go]: let's have consts for all of these durations as well.

File: agent/engine/engine_integ_test.go (hunk 1 above)
Comment (richardpen) [reply agent/engine/engine_integ_test.go]: I don't think there is a particular reason to not, these codes are all inherited from the existing integration test code. I will try to refactor it a little bit. The same for the other comments.

File: agent/engine/engine_integ_test.go [hunk 5]
```diff
@@ -125,6 +132,27 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
...
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return conn, err
+}
+
+func removeImage(t *testing.T, img string) {
+	client, err := docker.NewClient(endpoint)
+
```
Comment (aaithal) [inline agent/engine/engine_integ_test.go]: minor: redundant new line

File: agent/engine/engine_integ_test.go [hunk 6]
```diff
@@ -125,6 +132,27 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
...
+	require.NoError(t, err, "create docker client failed")
+
```
Comment (aaithal) [inline agent/engine/engine_integ_test.go]: minor: redundant new line

**Pull request merged**